pkg/resolvers/endpoints.go

```diff
@@ -185,14 +185,64 @@ func (r *defaultEndpointsResolver) resolveNetworkPeers(ctx context.Context, poli
 		} else {
 			namespaces = []string{policy.Namespace}
 		}
-		r.logger.V(1).Info("Namespaces for network peers resolution", "list", namespaces, "policy", k8s.NamespacedName(policy))
+
+		var portsToApply []policyinfo.Port
+		// populate the policy applied targets' ports
+		if policyType == networking.PolicyTypeIngress {
+			portsToApply = r.getIngressRulesPorts(ctx, policy.Namespace, &policy.Spec.PodSelector, ports)
+		}
+
 		for _, ns := range namespaces {
-			networkPeers = append(networkPeers, r.getMatchingPodAddresses(ctx, peer.PodSelector, ns, ports)...)
+			networkPeers = append(networkPeers, r.getMatchingPodAddresses(ctx, peer.PodSelector, ns, portsToApply, ports, policyType)...)
```

Comment:
If `portsToApply` is `nil` say for instance we hit the "Unable to List Pods" then why do even we need to go in the loop? Because eventually `portList := r.getPortList(pod, ports)` would be `0` and we just continue to the next pod isn't it? Am I missing anything?